File: java/sui-backend-app/src/main/java/ru/mos/sms/rest/payload/JwtAuthenticationResponse.java
package ru.mos.sms.rest.payload;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.smsoft.sui.suisecurity.security.UserPrincipal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JwtAuthenticationResponse {
    private String accessToken;
    private Long id;
    private String name;
    private String[] roles;

    public JwtAuthenticationResponse(String token, UserPrincipal principal) {
        this.accessToken = token;
        this.id = principal.getUser().getId();
        this.name = principal.getUser().getName();
        this.roles = principal.getAuthorities()
                .stream()
                .map(Object::toString)
                .map(s -> s.replace("ROLE_", ""))
                .toArray(String[]::new);
    }
}

File: java/sui-security/src/main/java/ru/smsoft/sui/suisecurity/model/JwtAuthenticationResponse.kt
package ru.smsoft.sui.suisecurity.model

import org.springframework.http.ResponseEntity
import ru.smsoft.sui.suisecurity.security.UserPrincipal

fun LoginResult.toResponseEntity() = ResponseEntity.ok(JwtAuthenticationResponse(this.jwt, this.principal))

class JwtAuthenticationResponse(
    val accessToken: String,
    principal: UserPrincipal
) {

    val id: Long = principal.user.id
    val name: String = principal.user.name
    val roles = principal.authorities.map { it.toString() }.map { it.replace("ROLE_", "") }.toTypedArray()

}
